lib/msf/ui/console/driver.rb

```diff
@@ -568,6 +570,33 @@ def handle_loglevel(val)
     set_log_level(Msf::LogSource, val)
   end
 
+  #
+  # This method monkeypatches Net::SSH's identification string.
+  #
+  def handle_sshversion(val)
+    return false unless val && val.kind_of?(String) && !val.empty?
+
+    begin
+      require 'net/ssh'
+    rescue LoadError
+      return false
+    end
+
```

Comment:
The only way you'll be able to get rid of the hacks is with a PR to https://github.com/net-ssh/net-ssh. Looks live enough!